test/e2e/deployment.go

```diff
@@ -187,7 +187,12 @@ func stopDeploymentMaybeOverlap(c clientset.Interface, internalClient internalcl
 	reaper, err := kubectl.ReaperFor(extensionsinternal.Kind("Deployment"), internalClient)
 	Expect(err).NotTo(HaveOccurred())
 	timeout := 1 * time.Minute
-	err = reaper.Stop(ns, deployment.Name, timeout, api.NewDeleteOptions(0))
+
+	// Make deployment go away immediately by removing
+	// the orphan finalizer
+	var falseVar = false
+	nonOrphanOption := api.DeleteOptions{OrphanDependents: &falseVar}
```

Comment:
Actually I think we could go ahead and set DeleteOptions in kubectl even in this PR. It won't be tricky if we don't want to handle different server versions. We will be able to drop the client-side reaper in 1.9 and use just the garbage collector.